src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java

```diff
@@ -140,16 +165,199 @@ private int calculateTotal(int base, TFloatList multipliers, TIntList modifiers)
+                }
+            }
+
+            characterSounds.lastSoundTime = time.getGameTimeInMs();
+            entity.saveComponent(characterSounds);
+
+        }
+    }
+
+    /**
+     * Causes damage to entity when fallingDamageSpeedThreshold is breached.
+     *
+     * @param event VerticalCollisionEvent sent when falling speed threshold is crossed.
+     * @param entity The entity which is damaged due to falling.
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onLand(VerticalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+        float speed = Math.abs(event.getVelocity().y);
+
+        if (speed > health.fallingDamageSpeedThreshold) {
+            int damage = (int) ((speed - health.fallingDamageSpeedThreshold) * health.excessSpeedDamageMultiplier);
+            if (damage > 0) {
+                checkDamage(entity, damage, EngineDamageTypes.PHYSICAL.get(), EntityRef.NULL, EntityRef.NULL);
+            }
+        }
+    }
+
+    /**
+     * Inflicts damage to entity horizontalDamageSpeedThreshold is breached.
+     *
+     * @param event HorizontalCollisionEvent sent when "falling horizontally".
+     * @param entity Entity which is damaged on "horizontal fall".
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onCrash(HorizontalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+
+        Vector3f vel = new Vector3f(event.getVelocity());
+        vel.y = 0;
+        float speed = vel.length();
+
+        if (speed > health.horizontalDamageSpeedThreshold) {
+            int damage = (int) ((speed - health.horizontalDamageSpeedThreshold) * health.excessSpeedDamageMultiplier);
+            if (damage > 0) {
+                checkDamage(entity, damage, EngineDamageTypes.PHYSICAL.get(), EntityRef.NULL, EntityRef.NULL);
+            }
+        }
+    }
+
+    /**
+     * Plays landing sound on crashing horizontally.
+     *
+     * @param event HorizontalCollisionEvent sent when "falling horizontally".
+     * @param entity Entity which is damaged on "horizontal fall".
+     * @param characterSounds For getting the sound to be played on crash.
+     * @param healthComponent To play sound only when threshold speed is crossed.
+     */
+    @ReceiveEvent
+    public void onCrash(HorizontalCollisionEvent event, EntityRef entity, CharacterSoundComponent characterSounds, HealthComponent healthComponent) {
+        Vector3f horizVelocity = new Vector3f(event.getVelocity());
+        horizVelocity.y = 0;
+        float velocity = horizVelocity.length();
+
+        if (velocity > healthComponent.horizontalDamageSpeedThreshold) {
+            if (characterSounds.lastSoundTime + CharacterSoundSystem.MIN_TIME < time.getGameTimeInMs()) {
+                StaticSound sound = random.nextItem(characterSounds.landingSounds);
```

Comment:
Similarly, there are a few segments of code where sounds are played, which could be extracted into a common private method

Reply:
Though this can be extracted, the `onCrash` only plays sound on the client (by sending PlaySoundEvent), but damage plays sound on both client and server.
So should the client sound way be extracted to a private method?